Dockerfile.location

```diff
@@ -1,5 +1,5 @@
 # go version here matches our go.mod
-FROM registry.redhat.io/rhel8/go-toolset:1.22.9 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.22.9 AS builder
```

Comment:
Switching to the public ubi9 builders saves me from having to set my registry.redhat.io token as a secret.

WDYT @gabemontero?

Reply:
yeah hopefully by the time we switch to konflux they will have sorted out pulling official images from the TBR

works for me